File: unidad-1/prac2-6/prac2-6.cpp
#include <iostream>
#include <vector>
#include <ctime>
#include <cstdlib>

using namespace std;

vector<vector<int>> generarMatriz(int filas, int columnas) {
    vector<vector<int>> matriz(filas, vector<int>(columnas));

    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < columnas; j++) {
            matriz[i][j] = rand() % 101;
        }
    }
    return matriz;
}

int contarPares(const vector<vector<int>>& matriz) {
    int conteo = 0;

    for (const auto& fila : matriz) {
        for (int val : fila) {
            if (val % 2 == 0) {
                conteo++;
            }
        }
    }
    return conteo;
}

int main() {
    srand(time(0));

    vector<pair<int,int>> tamanos = {
        {50,50},
        {100,100},
        {200,200}
    };

    for (auto tam : tamanos) {
        int filas = tam.first;
        int columnas = tam.second;

        vector<vector<int>> matriz = generarMatriz(filas, columnas);

        clock_t inicio = clock();
        int resultado = contarPares(matriz);
        clock_t fin = clock();

        double tiempo = double(fin - inicio) / CLOCKS_PER_SEC;

        cout << "Matriz: " << filas << "x" << columnas << endl;
        cout << "Numeros pares: " << resultado << endl;
        cout << "Tiempo de ejecucion: " << tiempo << " segundos" << endl;
        cout << "----------------------------------" << endl;
    }

    return 0;
}

File: unidad-1/prac2-6/prac2-6.py
import random
import time

def generar_matriz(filas, columnas):
    return [[random.randint(0, 100) for _ in range(columnas)] for _ in range(filas)]

def contar_pares(matriz):
    conteo = 0
    for fila in matriz:
        for valor in fila:
            if valor % 2 == 0:
                conteo += 1
    return conteo

# tamaños a evaluar
tamanos = [(50,50), (100,100), (200,200)]

for filas, columnas in tamanos:
    matriz = generar_matriz(filas, columnas)

    inicio = time.time()
    resultado = contar_pares(matriz)
    fin = time.time()

    print(f"Matriz: {filas}x{columnas}")
    print(f"Números pares: {resultado}")
    print(f"Tiempo de ejecución: {fin - inicio:.6f} segundos")
    print("-" * 40)